src/service.rs

```diff
@@ -1345,6 +1345,28 @@ impl Service {
                     }
                     InsertResult::ValueUpdated | InsertResult::UpdatedPending => {}
                     InsertResult::Failed(reason) => {
+                        // On large networks with limited IPv6 nodes, it is hard to get enough
+                        // PONG votes in order to estimate our external IP address. Often the
+                        // routing table can be full, and so we reject useful IPv6 here.
+                        //
+                        // If we are low on votes and we initiated this connection (i.e it was not
+                        // forced on us) then lets get a PONG from this node.
+
+                        if direction == ConnectionDirection::Outgoing {
+                            if let Some(ip_votes) = self.ip_votes.as_mut() {
+                                match (ip_votes.majority(), enr.udp4_socket(), enr.udp6_socket()) {
+                                    // We don't have enough ipv4 votes, but this is an IPv4 node.
+                                    ((Some(_), None), Some(_), _) |
```

Comment:
Since `ip_votes.majority()` returns `(Option<SocketAddrV4>, Option<SocketAddrV6>)`, should this look like this?

```suggestion
                                    ((None, Some(_)), Some(_), _) |
```

Reply:
ahah yeah. I seemed to have confused myself in the last commit. I reversed these. I'll change it back.